### PR TITLE
[chore] use functions from Utils.String module for to/from List Bits8

### PR DESCRIPTION
**Pull request by jarlah**
following the latest changes introduced by #13

### CODE REVIEW

File: tests/src/AuthorizationTest.idr [hunk 1]
```diff
@@ -16,17 +17,20 @@ test_base64_encoding_decoding =
...
-            passwd'    = last userpasswd
-          in case basicauth == "Basic YWxhZGRpbjpvcGVuc2VzYW1l" &&
-                  user      == user'                            &&
-                  passwd    == passwd' of
-            True  =>
-              pure ()
-            False =>
-              throwE "base64 encode/decode error: expected \{user} and \{user'}, \{passwd} and \{passwd'} to be equivalent."
+        case utf8_pack basicauth'' of
+          Nothing =>
```
Comment (jarlah) [inline tests/src/AuthorizationTest.idr]: hmm should be a function, a test helper, thats aptly named justOrFail or something

**Comment by jarlah** [comment]
@Matthew-Mosior what do you think? More readable? I got sore eyes from looking at these List bits8 mappings operations. And i flattened the test too.

File: tests/src/AuthorizationTest.idr (hunk 1 above)
Comment (jarlah) [reply tests/src/AuthorizationTest.idr]: starts getting unwieldy when you have several layers of case matches with Nothings ....

**Comment by Matthew-Mosior** [comment]
> @Matthew-Mosior what do you think? More readable? I got sore eyes from looking at these List bits8 mappings operations. And i flattened the test too.

This looks great, I really like this.

**Review by Matthew-Mosior — APPROVED** [review]
lgtm!

**Pull request merged**